luaui/Include/blueprint_substitution/definitions.lua

```diff
@@ -0,0 +1,166 @@
+-- luaui/Include/blueprint_substitution/definitions.lua
```

Comment:
![image](https://github.com/user-attachments/assets/55971f3b-84c9-448e-ba71-a2260cb88269)
Not entirely sold on this factoring/naming. I could see us wanting to call definitions "category_definitions" and logic "blueprint_substitution_logic", for easier finding. This was really just me doing the factoring I'm used to/prefer, which avoids redundancy in filenames. For logic and definitions, I could also see us just embedding these directly in a gadget? Just let me know I didn't spend too much time thinking about this particular problem.